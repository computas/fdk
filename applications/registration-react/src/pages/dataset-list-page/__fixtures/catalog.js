export default {
  item: {
    id: '910244132',
    uri: 'http://brreg.no/catalogs/910244132',
    title: {
      nb: 'Datakatalog for RAMSUND OG ROGNAN REVISJON'
    },
    description: {},
    publisher: {
      uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
      name: 'RAMSUND OG ROGNAN REVISJON'
    }
  }
};
