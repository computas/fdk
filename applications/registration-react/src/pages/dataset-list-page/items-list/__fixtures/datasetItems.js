export const datasetItems = {
  isFetchingDatasets: false,
  datasetItems: [
    {
      id: '084f747f-04a9-460e-b07e-ee74c60a2086',
      uri:
        'http://brreg.no/catalogs/910244132/datasets/084f747f-04a9-460e-b07e-ee74c60a2086',
      title: {
        nb: 'test ois'
      },
      description: {
        nb: 'tester'
      },
      objective: {
        nb: ''
      },
      publisher: {
        uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
        name: 'RAMSUND OG ROGNAN REVISJON'
      },
      type: 'Data',
      landingPage: [''],
      theme: [
        {
          uri: '',
          title: {
            nb: ''
          }
        },
        {
          uri:
            'http://publications.europa.eu/resource/authority/data-theme/AGRI',
          title: {
            nb: 'Jordbruk, fiskeri, skogbruk og mat'
          }
        }
      ],
      accessRights: {
        uri:
          'http://publications.europa.eu/resource/authority/access-right/NON_PUBLIC',
        prefLabel: {}
      },
      legalBasisForRestriction: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'skjerming'
          },
          extraType: null
        }
      ],
      legalBasisForProcessing: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'behandling'
          },
          extraType: null
        }
      ],
      legalBasisForAccess: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'utlevering'
          },
          extraType: null
        }
      ],
      catalogId: '910244132',
      _lastModified: '2018-05-07T10:36:20.388+0000',
      registrationStatus: 'PUBLISH',
      subject: [
        {
          uri: 'https://data-david.github.io/Begrep/begrep/Enhet',
          identifier: 'https://data-david.github.io/Begrep/begrep/Enhet',
          prefLabel: {
            no: 'enhet'
          },
          altLabel: null,
          definition: {
            no: 'alt som er registrert med et organisasjonsnummer '
          },
          note: {
            no:
              'Alle hovedenheter, underenheter og organisasjonsledd som er identifisert med et organisasjonsnummer.'
          },
          source: 'https://jira.brreg.no/browse/BEGREP-208',
          creator: {
            uri: 'http://data.brreg.no/enhetsregisteret/enhet/974760673'
          },
          inScheme: ['http://data-david.github.io/vokabular/Befolkning'],
          datasets: null
        }
      ],
      keyword: [
        {
          nb: 'søkeord'
        }
      ],
      contactPoint: [
        {
          email: 'epost@kontakskjema.no',
          hasTelephone: '12345678',
          hasURL: 'https://www.google.no',
          organizationUnit: 'Kontaktpunkt'
        }
      ],
      conformsTo: [
        {
          uri: '',
          prefLabel: {
            nb: 'test'
          },
          extraType: null
        }
      ],
      references: [
        {
          referenceType: {
            uri: 'dct:isPartOf',
            code: 'isPartOf',
            prefLabel: {}
          },
          source: {
            uri:
              'http://brreg.no/catalogs/910244132/datasets/0aa1861a-9af0-4388-a81e-6b05de0fb63e',
            prefLabel: null,
            extraType: null
          }
        }
      ],
      hasRelevanceAnnotation: {
        inDimension: 'iso:Relevance',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Relevans'
        }
      },
      hasCompletenessAnnotation: {
        inDimension: 'iso:Completeness',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Kompletthet'
        }
      },
      hasAccuracyAnnotation: {
        inDimension: 'iso:Accuracy',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Nøyaktighet'
        }
      },
      hasAvailabilityAnnotation: {
        inDimension: 'iso:Availability',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Tilgjengelighet'
        }
      },
      distribution: [
        {
          id: '',
          description: {
            nb: 'Beskrivelse av distribusjon'
          },
          accessURL: ['http://google.no'],
          license: {
            uri: 'http://creativecommons.org/publicdomain/zero/1.0/',
            prefLabel: null,
            extraType: null
          },
          conformsTo: [
            {
              prefLabel: {
                nb: 'Standard'
              },
              uri: 'http://google.no'
            }
          ],
          page: [
            {
              uri: 'https://google.com',
              prefLabel: {
                nb: ''
              },
              extraType: null
            }
          ],
          format: ['json', 'xml'],
          type: 'API'
        }
      ],
      sample: [
        {
          id: '',
          description: {
            nb: ''
          },
          accessURL: [],
          license: {
            uri: '',
            prefLabel: {
              nb: ''
            },
            extraType: null
          },
          conformsTo: [],
          page: [
            {
              uri: '',
              prefLabel: {
                nb: ''
              },
              extraType: null
            }
          ],
          format: [],
          type: 'API'
        }
      ],
      openLicenseItems: [
        {
          uri: 'http://creativecommons.org/licenses/by/4.0/',
          code: 'CC BY 4.0',
          prefLabel_no: 'Creative Commons Navngivelse 4.0 Internasjonal'
        },
        {
          uri: 'http://creativecommons.org/publicdomain/zero/1.0/',
          code: 'CC0 1.0',
          prefLabel_no: 'Creative Commons Universal Fristatus-erklæring'
        },
        {
          uri: 'http://data.norge.no/nlod/',
          code: 'NLOD',
          prefLabel_no: 'Norsk lisens for offentlige data'
        }
      ],
      informationModel: [
        {
          uri: 'http://google.no',
          prefLabel: {
            nb: 'test'
          },
          extraType: null
        }
      ],
      frequencyItems: null,
      provenanceItems: null,
      provenance: {},
      modified: null,
      hasCurrentnessAnnotation: {
        inDimension: 'iso:Currentness',
        motivatedBy: 'dqv:qualityAssessment',
        hasBody: {
          nb: 'Aktualitet'
        }
      },
      accrualPeriodicity: {
        uri:
          'http://publications.europa.eu/resource/authority/frequency/ANNUAL',
        code: 'ANNUAL',
        prefLabel: {}
      },
      spatial: [
        {
          uri: 'Oslo'
        }
      ],
      temporal: [{}],
      issued: null,
      language: []
    }
  ]
};
