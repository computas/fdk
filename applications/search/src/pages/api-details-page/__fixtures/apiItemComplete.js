export const apiItemComplete = {
  uri: 'jar:file:/app.jar!/BOOT-INF/classes!/datakatalog-api.json',
  title: {
    no: 'National Data Directory Search API'
  },
  description: {
    no:
      'Provides a basic search api against the National Data Directory of Norway'
  },
  contactPoint: [
    {
      uri: 'https://fellesdatakatalog.brreg.no',
      email: 'fellesdatakatalog@brreg.no',
      organizationName: 'Brønnøysundregistrene'
    }
  ],
  openApi: {
    openapi: null,
    info: {
      description:
        'Provides a basic search api against the National Data Directory of Norway',
      version: '1.0',
      title: 'National Data Directory Search API',
      termsOfService: 'https://fellesdatakatalog.brreg.no/about',
      contact: {
        name: 'Brønnøysundregistrene',
        url: 'https://fellesdatakatalog.brreg.no',
        email: 'fellesdatakatalog@brreg.no'
      },
      license: {
        name: 'License of API',
        url: 'http://data.norge.no/nlod/no/2.0'
      }
    },
    servers: null,
    paths: {
      '/catalogs': {
        get: {
          description:
            'The three formats are: text/turtle, application/ld+json and application/rdf+xml',
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/harvest/catalog': {
        get: {
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/catalogs/datasets': {
        get: {
          description:
            'The three formats are: text/turtle, application/ld+json and application/rdf+xml',
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/publisher/hierarchy': {
        get: {
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/datasets': {
        get: {
          description:
            'Returns a list of matching datasets wrapped in a elasticsearch response. Max number returned by a single query is 100. Size parameters greater than 100 will not return more than 100 datasets. In order to access all datasets, use multiple queries and increment from parameter.',
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/datasets/**': {
        get: {
          description: "You must specify the dataset's identifier",
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/publisher': {
        get: {
          description:
            'Returns the elasticsearch response with matching publishers',
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/aggregateDataset': {
        get: {
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      },
      '/terms': {
        get: {
          description:
            'Returns the elasticsearch response with matching terms (dct:subject)',
          responses: {
            '200': {
              description: 'OK'
            },
            '401': {
              description: 'Unauthorized'
            },
            '403': {
              description: 'Forbidden'
            },
            '404': {
              description: 'Not Found'
            }
          }
        }
      }
    }
  }
};
