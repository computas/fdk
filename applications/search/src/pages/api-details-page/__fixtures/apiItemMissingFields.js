export const apiItemMissingFields = {
  uri: 'jar:file:/app.jar!/BOOT-INF/classes!/datakatalog-api.json',
  openApi: {
    openapi: null,
    info: {
      description:
        'Provides a basic search api against the National Data Directory of Norway',
      version: '1.0',
      title: 'National Data Directory Search API',
      termsOfService: 'https://fellesdatakatalog.brreg.no/about',
      contact: {
        name: 'Brønnøysundregistrene',
        url: 'https://fellesdatakatalog.brreg.no',
        email: 'fellesdatakatalog@brreg.no'
      },
      license: {
        name: 'License of API',
        url: 'http://data.norge.no/nlod/no/2.0'
      }
    },
    servers: null
  }
};
