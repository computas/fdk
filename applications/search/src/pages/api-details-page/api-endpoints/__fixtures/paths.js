export const paths = {
  '/catalogs': {
    get: {
      description:
        'The three formats are: text/turtle, application/ld+json and application/rdf+xml',
      responses: {
        '200': {
          description: 'OK'
        },
        '401': {
          description: 'Unauthorized'
        },
        '403': {
          description: 'Forbidden'
        },
        '404': {
          description: 'Not Found'
        }
      }
    }
  }
};
