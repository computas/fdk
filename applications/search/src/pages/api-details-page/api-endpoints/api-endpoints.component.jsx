import React from 'react';
import PropTypes from 'prop-types';

import localization from '../../../lib/localization';

export const renderPathMethod = (path, method, methodDeclaration) => (
  <React.Fragment key={`${method}-${path}`}>
    <div className="row">
      <div className="col-5 fdk-text-strong">
        {method.toUpperCase()} {path}
      </div>
      <div className="col-7">{methodDeclaration.description}</div>
    </div>
    <hr />
  </React.Fragment>
);

export const renderPath = (path, declaration) =>
  declaration &&
  Object.keys(declaration).map(method =>
    renderPathMethod(path, method, declaration[method])
  );

export const renderPaths = paths => {
  const renderedMethods =
    paths && Object.keys(paths).map(path => renderPath(path, paths[path]));
  return <div>{renderedMethods}</div>;
};

const renderLinks = (apiSpecUrl, apiDocUrl) => {
  if (apiSpecUrl || apiDocUrl) {
    return (
      <div className="row">
        <div className="col-12">
          {apiDocUrl && (
            <a href={apiDocUrl} className="mr-5">
              {localization.documentation}
              <i className="fa fa-external-link fdk-fa-right" />
            </a>
          )}
          {apiSpecUrl && (
            <a href={apiSpecUrl} className="mr-5">
              {localization.specification}
              <i className="fa fa-external-link fdk-fa-right" />
            </a>
          )}
        </div>
      </div>
    );
  }
  return null;
};

export const ApiEndpoints = props => {
  const { name, paths, apiSpecUrl, apiDocUrl } = props;
  return (
    <section className="mb-5 list-type1" name={name}>
      <div className="row">
        <div className="col-5">
          <h3 className="">{localization.api.endpoints.operation}</h3>
        </div>
        <div className="col-7">
          <h3 className="">{localization.api.endpoints.description}</h3>
        </div>
      </div>
      <hr />
      {renderPaths(paths)}
      {renderLinks(apiSpecUrl, apiDocUrl)}
    </section>
  );
};

ApiEndpoints.defaultProps = {
  name: null,
  paths: null,
  apiSpecUrl: null,
  apiDocUrl: null
};

ApiEndpoints.propTypes = {
  name: PropTypes.string,
  paths: PropTypes.object,
  apiSpecUrl: PropTypes.string,
  apiDocUrl: PropTypes.string
};
