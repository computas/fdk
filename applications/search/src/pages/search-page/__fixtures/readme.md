apisApiResponse.json is generated as:

curl "https://fellesdatakatalog.brreg.no/api-cat/api/search?q=barnehage&size=50" -o applications/search/src/pages/search-page/__fixtures/apisApiResponse.json

datasetsApiResponse.json is generated as:
curl "https://fellesdatakatalog.brreg.no/datasets?q=Marka&size=50" -H "Accept: application/json" -o applications/search/src/pages/search-page/__fixtures/datasetsApiResponse.json
