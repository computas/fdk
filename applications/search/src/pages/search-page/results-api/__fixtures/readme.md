apisApiResponse.json is generated as:

curl "http://localhost:8080/api-cat/api/search?q=a&size=50" -o apisApiResponse.json