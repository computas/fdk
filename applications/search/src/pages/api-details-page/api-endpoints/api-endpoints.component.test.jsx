import React from 'react';
import { shallow } from 'enzyme';
import { ApiEndpoints } from './api-endpoints.component';
import { paths } from './__fixtures/paths';

let defaultProps;

test('should render api-ApiEndpoints correctly with no props', () => {
  const wrapper = shallow(<ApiEndpoints />);
  expect(wrapper).toMatchSnapshot();
});

test('should render api-ApiEndpoints correctly', () => {
  defaultProps = {
    name: 'Test',
    paths
  };
  const wrapper = shallow(<ApiEndpoints {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
