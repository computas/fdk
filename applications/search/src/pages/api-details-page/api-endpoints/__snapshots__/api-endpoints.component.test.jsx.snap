// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render api-ApiEndpoints correctly 1`] = `
<section
  className="mb-5 list-regular"
  name="Test"
>
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-5 pl-0"
    >
      <h3
        className=""
      >
        Operation
      </h3>
    </div>
    <div
      className="col-7"
    >
      <h3
        className=""
      >
        Description
      </h3>
    </div>
  </div>
  <React.Fragment
    key="get-/catalogs"
  >
    <div
      className="d-flex list-regular--item"
    >
      <div
        className="col-5 pl-0 fdk-text-strong"
      >
        GET
         
        /catalogs
      </div>
      <div
        className="col-7"
      >
        The three formats are: text/turtle, application/ld+json and application/rdf+xml
      </div>
    </div>
  </React.Fragment>
</section>
`;

exports[`should render api-ApiEndpoints correctly with no props 1`] = `
<section
  className="mb-5 list-regular"
  name={null}
>
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-5 pl-0"
    >
      <h3
        className=""
      >
        Operation
      </h3>
    </div>
    <div
      className="col-7"
    >
      <h3
        className=""
      >
        Description
      </h3>
    </div>
  </div>
</section>
`;
